### ISSUE TITLE
ldm model question

### ISSUE THREAD

**Issue by yangjie1874** [UTokyo-FieldPhenomics-Lab/DODA#2]
Hello author, I am amazed by your work. When I trained the generative model, I used the LDM method to train 12,000 416×416 images without adding pre-trained weights. When generating images, why are all the generated images noisy?

**Comment by illrayy** [comment]
Thank you for your interest in our work. Generating noise can be caused by many different reasons, one possible reason is the dataset too small or the iterations not enough that the model hasn't converged yet. When you training, are the images in the 'image_log' folder that start with 'samples_gs_' also noise?

**Comment by yangjie1874** [comment]
I followed your suggestion and found that the 'image_log' folder that starts with 'samples_gs_' is not noise, but the subsequent images will have noise

**Comment by illrayy** [comment]
‘samples_gs_’ is the normal output of the model. If ‘samples_gs_’ are meaningful images instead of noise, it means that the model already has the ability to generate images.
What are the file names of the noisy "subsequent images"? Also, did you only train ldm and not controlnet? And which script did you use to generate the images?

**Comment by yangjie1874** [comment]
Hello, the script I use is generate_data_for_target_domain.py，on my own dataset, I first use train_wheat.py to train the diffusion model, and then use generate_data_for_target_domain.py to generate images using the weights of the diffusion model. The configuration file is DODA_wheat_ldm_kl_4.yaml, ckpt_path: "coco-128.ckpt", this is my process

**Comment by yangjie1874** [comment]
I thought that using ldm did not require controlnet training

**Comment by illrayy** [comment]
yes using ldm not requires controlnet, and your process also looks good.

The images in the 'image_log' folder record the training process. The file names of the images represent how the images were obtained, and the number after 'gs' represents the global step. If your images look like the following examples, then the training process is correct.

> "inputs_gs-200000_e-000069_b-002246"

<img width="600" alt="inputs_gs-200000_e-000069_b-002246" src="https://github.com/UTokyo-FieldPhenomics-Lab/DODA/assets/93829503/83016d17-06a9-482f-95e6-644ab4ee4668">

> "reconstruction_gs-200000_e-000069_b-002246", obtained by using VAE to encode and decode input image.

<img width="600" alt="reconstruction_gs-200000_e-000069_b-002246" src="https://github.com/UTokyo-FieldPhenomics-Lab/DODA/assets/93829503/317cc24d-08f7-4140-99d5-d7c44225bab6">

> "progressive_row_gs-200000_e-000069_b-002246", strat form the noise, denoising images step by step 

<img width="600" alt="progressive_row_gs-200000_e-000069_b-002246" src="https://github.com/UTokyo-FieldPhenomics-Lab/DODA/assets/93829503/4a43c7e5-4c8e-4f13-9901-c9fa66c9cce2">


> "samples_gs-200000_e-000069_b-002246", output images

<img width="600" alt="samples_gs-200000_e-000069_b-002246" src="https://github.com/UTokyo-FieldPhenomics-Lab/DODA/assets/93829503/b499e749-7070-4641-ac9c-0a573f8d0772">

**Comment by illrayy** [comment]
For the generate_data_for_target_domain.py and the config file, did you make any changes?

**Comment by yangjie1874** [comment]
Thank you very much for your patience，I trained on my own dataset，in your example， start with reconstruction-gs-xxx-xxx-xxx，subsequent image examples are all noise，For the generate_data_for_target_domain.py and the config file，i just adjusted dataset path and my weights, i think the problem is train_![image](https://github.com/UTokyo-FieldPhenomics-Lab/DODA/assets/143858218/c0ac242c-1337-4082-a9e8-2c4ce4c41e00),  
![image](https://github.com/UTokyo-FieldPhenomics-Lab/DODA/assets/143858218/99cb2a0a-511b-4b82-9882-d558670c157b)

![123](https://github.com/UTokyo-FieldPhenomics-Lab/DODA/assets/143858218/79d9baeb-a34f-45ce-89cb-aaf49bef2402)

**Comment by illrayy** [comment]
yes, ldm uses vae to encode image into latent space
Sorry I didn't upload a separate vae before because the weights for this part are included in DODA-ldm, you can use the vae weight here: [https://drive.google.com/file/d/1XHmtZR95uSbFcY-y6wCffgV5uUM1x8pC/view?usp=sharing](url), or train a vae by yourself using configs\autoencoder\DODA_wheat_autoencoder_kl_64x64x3.yaml as config.

**Comment by yangjie1874** [comment]
thank so much!!!, by the way, do I need to re-train VAE on my own dataset?, or just use yours

**Comment by illrayy** [comment]
In the original LDM paper (and early versions of stable diffusion), for different datasets, the VAEs used were all trained on OpenImage, so if your dataset is not very different from Global Wheat, you can use the pre-trained VAE, or finetuning the model

**Comment by yangjie1874** [comment]
Hello, as shown in the figure, in the image generated by generate_data_for_target_domain.py, the label position I want is very random and almost wrong. I want to know the reason
![dce3b4ab28a1d22d611633314237da2d](https://github.com/UTokyo-FieldPhenomics-Lab/DODA/assets/143858218/c4ed645c-2f38-495c-9c31-4579b8d57cf8)
![fc5746b9270ee68ed77d1b3334268fd1](https://github.com/UTokyo-FieldPhenomics-Lab/DODA/assets/143858218/28af6e2f-e736-44ef-b1e5-7d2a5d6fc437)

**Comment by yangjie1874** [comment]
I mean there is nothing wrong with the image, but the label doesn't match

**Comment by illrayy** [comment]
That's a differnet question. please describe it in this issue #3
